packages/python-editor/src/ScriptEditor.tsx

```diff
@@ -85,15 +92,20 @@ export class PythonFileEditor extends DocumentWidget<
     options: DocumentWidget.IOptions<FileEditor, DocumentRegistry.ICodeModel>
   ) {
     super(options);
-    this.addClass(PYTHON_FILE_EDITOR_CLASS);
+    this.addClass(SCRIPT_EDITOR_CLASS);
     this.model = this.content.model;
-    this.runner = new PythonRunner(this.model, this.context, this.disableRun);
-    this.kernelName = null;
+    this.runner = new ScriptRunner(this.disableRun);
+    this.kernelSelectorRef = null;
+    this.kernelName = options.context.sessionContext.kernelPreference.language;
     this.emptyOutput = true;
     this.runDisabled = false;
+    this.controller = new ScriptEditorController();
+    this.editorLanguage = this.kernelName.toLowerCase().includes(PYTHON)
+      ? PYTHON
+      : R;
 
-    // Add python icon to main tab
-    this.title.icon = pyIcon;
+    // Add icon to main tab
+    this.title.icon = this.editorLanguage === PYTHON ? pyIcon : rIcon;
```

Comment:
I agree, making subclasses would make these language-specific properties better contained.
This was the very first implementation idea, but we decided in handling it all in one class for now. 
The editor language is defined by the file type specified for the script editor factory, and since we are only creating script editors for Python or R files, the Script editor language assumptions can be considered safe, only not the most elegant. 😅  
We can revisit the implementation on a follow up PR.